tracer.go

```diff
@@ -73,6 +73,12 @@ type Options struct {
 	// When set, it attempts to exacerbate issues emanating from use of Spans
 	// after calling Finish by running additional assertions.
 	DebugAssertUseAfterFinish bool
+	// DisableSpanPool disables reuse of the underlying span structs.
+	// Buyer Beware, any operations called on a span after Finish is undefined
+	// behavior. The main usecase of disabling the pool is to allow the creation
```

Comment:
Yes, we shouldn't break any code, that limits the options. 

But I would say the race condition is worse than always resetting the span immediately. I.e., if the pool is enabled the spans should be reset in finish() to avoid a situation where things sometimes work and sometimes do not.